### ISSUE TITLE
Why not PR into the main havoc?

### ISSUE THREAD

**Issue by eried** [jamesshao8/portapack-havoc-modified#7]
I wonder why you havent integrated this into the main repo

**Comment by jamesshao8** [comment]
I wrote to the creator of havoc once about that. But I didnt get the reply. So I didn't merge.
My code here shouldn't be too old from the main repo. If otherwise, you can always compare the difference and combine them manually.
I have an instruction of what I modified and how I did that.
https://blog.csdn.net/shukebeta008/category_9718353.html

**Comment by eried** [comment]
You wrote about? he did not accepted the merge of a pull request?

**Comment by jamesshao8** [comment]
I just wrote an email to him. I didn't do the merge request through github yet.

"I have developed some new features (gps transmission, analog tv demodulation and BLE decoding). They are working but still needs refinement. I am wondering if I could merge into your repo, so that more users in this community share with  me their opinion."

That's what I wrote.

**Comment by eried** [comment]
Oh, you probably should do a PR and some of us will add some positive comments. He might be overwhelmed already (you know how it goes after a couple of years of a non paying project) I was actually thinking if is time for a new fresher fork... I want to make a simple app for "finding signals" (like a lost drone) but it would be a bummer if it never gets into the main fw

**Comment by jamesshao8** [comment]
What signal is sending out from your lost drone? Analog video or digital video like DJI?
I am also a drone developer. Most of the app I wrote for portapack is drone related. Analog video is used to watch analog fpv signal. BTLE and nrfl24l01(in future) decoding is to receive telemetry or remote control data.
For your application, I think you can try with kerberossdr, it will find the direction and location of the target signal. The only problem is the highest frequency it supports is 1.7GHz.

**Comment by RChadwick7** [comment]
I vote for backing a new fork. Furtek has recently admitted he has no desire to continue the project. There's a LOT of awesome activity over here. :)

**Comment by eried** [comment]
> What signal is sending out from your lost drone?

Oh, I was thinking just on a VMU signal level like the Ghostbusters' _P.K.E. Meter_ :P I just recently got my portapack and I was just learning about it. Trying to add some debug apps back from Sharebrain's repo I noticed that they were just commented out on havoc for no reason. Also it is kinda lame than even simple PR dont get any activity https://github.com/furrtek/portapack-havoc/pull/324 makes me feel the project is dead and disencourages other people from helping out.



> I vote for backing a new fork. Furtek has recently admitted he has no desire to continue the project. There's a LOT of awesome activity over here. :)

I would agree if @furrtek said that.

**Comment by furrtek** [comment]
`for no reason`: Already explained, space saving and preventing users from asking questions about something they shouldn't be using anyways. Debug functions -> for devs. Devs would know how to re-enable those functions.

`kinda lame than even simple PR dont get any activity`: Might be surprising, but I have other things to do which are actually much more rewarding than this. If you're mad because your 1-char fix took a few days to be merged, this isn't for you.

`There's a LOT of awesome activity over here`: Tens of people asking for stuff and nearly zero significant code contribution isn't what I would call "awesome". Having ideas and fixing two values in a .h is cute but doesn't get new stuff done.

In short: Want things to happen ? Simple: less talk, less emotions, more doing. If people finally get their hands dirty instead of pulling arms, that's great ! But be warned: there's a LOT of arm-pullers.

**Comment by eried** [comment]
Hey, thanks for merging my PR, I know it is simple but took me a few to find the reason in a project I dont know yet. BTW, I don’t think hostility will get people willing to collaborate. I tagged you because the quote regarding the no desire of continuing your fork, that's all.

**Comment by furrtek** [comment]
Got it.

**Comment by ytsqkl** [comment]
Everyone's time is limited. They usually have family life and work. All the projects they do are completed with rest time. This is a kind of dedication. Thank you very much, furrtek It can bring us the version of havoc. At the same time, I'd like to thank James shao8 for bringing us GPS, analog video and other functions. This is a public welfare project without any reward. Everyone is dedicated to hobbies. James shao8 has done a lot of videos and microblogs in China to teach you how to use and program. Open source program is to integrate the global technical force to enhance the system function,

**Comment by eried** [comment]
Anyways @jamesshao8, are you going to do a PR with your mods? 

I have been designing a 3d printable case for my hackrf :P but I am already done with that! so... I can  do a pull request to havoc with your features if you want

**Comment by jamesshao8** [comment]
Hi, I am not very familiar with github yet. You can do the PR on your side. Just remember to keep my copyright information on top of the code that I have written.

**Comment by eried** [comment]
OK, then probably the easiest would be is that you go to https://github.com/furrtek/portapack-havoc and click Fork. On that new Fork (that you can set to private) go to Settings / Manage access and add me in "Invite colaborator". In that clean fork I create 1 new branch per each feature and pull those to the original havoc and we can test it together

**Comment by jamesshao8** [comment]
OK. I finished doing what you suggested. Please check if everything is fine.

**Comment by eried** [comment]
@jamesshao8 can you enable issues in Settings in the new fork? so we can discuss there. 

https://github.com/jamesshao8/portapack-havoc/tree/analog-tv <- I did a cleanup and it is almost ready for the PR, however could you share a screenshot of it working? I do not have any way to generate a PAL signal :P I tried with my drone but the image is not clear:
![image](https://user-images.githubusercontent.com/1091420/79611195-5a192c80-80fa-11ea-9a0c-98c9c86ae229.png)

**Comment by jamesshao8** [comment]
Are you sure you are sending PAL? it looks like NTSC to me.
https://www.youtube.com/watch?v=VxBReLPbfZc
Here's a video. In the first part I watched analog tv in China which is pal. The second part is when I was receiving video from my drone which is ntsc, which is close to your result.
The signal strength from drone is usually strong, so you might need to decrease the lna value (40 in your figure), and adjust the vga value (30) so that IQ waveform dosen't saturate. You can use the time scope above the image to measure.
There are another 2 parameters for fine tune. The ppm on the upper-right of the screen (0) can be tuned so that the image stop rolling horizontally, and there is a hidden number above time scope, can be used to move image horizontally. This is just for fine tune. 
By the way, you can use hacktv as a signal generator. It's also on github. It can sends out a video in any analog tv standard through hackrf. Installing it on ubuntu 18.04 is easy.

How to enable issues? Most of the repo is enabled by default, I dont know why it isn't in the new fork. I didn't disable it manually.

**Comment by eried** [comment]
In settings, features. It was disabled by default I think on this particular repo. I wanted to continue discussing there to close this issue :)
